core/src/sync/message/status.rs

```diff
@@ -42,32 +37,60 @@ impl Handleable for Status {
 
         let mut latest: HashSet<H256> =
             self.terminal_block_hashes.iter().cloned().collect();
-        latest.extend(
-            ctx.manager.graph.initial_missed_block_hashes.lock().drain(),
-        );
-
-        let peer_state = SynchronizationPeerState {
-            id: ctx.peer,
-            protocol_version: self.protocol_version,
-            genesis_hash,
-            best_epoch: self.best_epoch,
-            latest_block_hashes: latest,
-            received_transaction_count: 0,
-            need_prop_trans: true,
-            notified_mode: None,
-            heartbeat: Instant::now(),
-        };
-
-        debug!(
-            "New peer (pv={:?}, gh={:?})",
-            self.protocol_version, self.genesis_hash
-        );
-
-        debug!("Peer {:?} connected", ctx.peer);
-        ctx.manager.syn.peer_connected(ctx.peer, peer_state);
-        ctx.manager.request_manager.on_peer_connected(ctx.peer);
-
-        ctx.manager.start_sync(ctx.io);
+
+        if let Ok(peer_info) = ctx.manager.syn.get_peer_info(&ctx.peer) {
+            let latest_updated = {
+                let mut peer_info = peer_info.write();
+                if peer_info.protocol_version != self.protocol_version {
```

Comment:
I think the protocol version is useless in protocol handler, since the session handshake already compared the protocol and its version.